### PR TITLE
Switch Tabs components to use React Native Pressable

### PR DESCRIPTION
**Pull request by Saadnajmi**
### Platforms Impacted
- [ ] iOS
- [x] macOS
- [x] win32 (Office)
- [x] windows
- [ ] android

### Description of changes

Following up from #1070, let's move over components off of `useAsPressable` / our fork of Pressable and onto Pressable from React Native, which contains the newer support for pointer events / android ripple / codegen commands.

### Verification

Hover, Focus, and Press work as expected on macOS. E2E testing on CI should confirm as much for windows/win32, but I'll try and test it out myself too.

https://user-images.githubusercontent.com/6722175/191470443-2362ba5c-305d-43f0-8ed5-071848617f54.mov

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [ ] Documentation and examples
- [x] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/experimental/Tabs/src/TabsItem.types.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export interface TabsItemInfo {
 }
 
 export interface TabsItemSlotProps {
-  root: React.PropsWithRef<IViewProps>;
+  root: React.PropsWithRef<PressableProps>;
```
Comment (Saadnajmi) [inline packages/experimental/Tabs/src/TabsItem.types.ts:149]: Technically we lose the adapter props here. Will have to think about that. Could probably easily make an "IPressableProps" if I _really_ wanted to.

Comment (rurikoaraki) [reply packages/experimental/Tabs/src/TabsItem.types.ts:149]: Yeah in practice as long as the props we expose for the component stay constant I think we're ok... Even if props that don't match the type are passed in, they're still on the object so they should get passed through, they just will make TS angry if someone tries to modify the root slot directly and add props that don't exist on PressableProps

Comment (rurikoaraki) [reply packages/experimental/Tabs/src/TabsItem.types.ts:149]: So really it'll be more of a problem for us than for clients.

**Comment by rurikoaraki** [comment]
I can test it for win32/windows for you.

File: packages/experimental/Tabs/src/TabsItem.types.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Saadnajmi) [reply packages/experimental/Tabs/src/TabsItem.types.ts:149]: Oh... That's probably what I'm missing :D

[the remark this answers is not in the document]
Comment (Saadnajmi) [reply packages/experimental/Tabs/src/TabsItem.types.ts:149]: Fixed

Comment (Saadnajmi) [reply packages/experimental/Tabs/src/TabsItem.types.ts:149]: I think type safety is still a but of an open problem in FURN. Adapters is a step towards it but if we enable platforms specific type safety checking via rex-kit, there's still issues. I'll try to spend a day thinking about how best to not regress the experience for ourselves.

**Review by rurikoaraki — APPROVED** [review]

**Pull request merged**